index.js

```diff
@@ -21,6 +21,53 @@
 import path from 'path';
 import semver from 'semver';
 import { execSync } from 'child_process';
+import { existsSync } from 'fs';
+import { EOL } from 'os';
+
+import userInput from 'commander';
+
+// This is a workaround for an issue in commander's argument parser.
+// See https://github.com/tj/commander.js/issues/512
+const getNormalisedArgs = () => {
+  const args = process.argv;
+
+  // Detect execution without a script (e.g. node -e <input>)
+  // and insert a second argument when appropriate.
+  // For more info about the defaults, see https://nodejs.org/docs/latest-v8.x/api/process.html#process_process_argv
+  if (!process.mainModule) {
+    const execPath = args.shift();
+    args.unshift(execPath, module.filename);
+  }
+
+  return args;
+};
+
+userInput
+  .option(
+    '-i, --ignore-local-bin',
+    'Ignore any binaries in ./node_modules/.bin',
+  )
+  .option('-v, --verbose', 'Prints information about all the binaries detected')
+  .parse(getNormalisedArgs());
+
+if (!userInput.verbose) {
+  console.log = () => {};
```

Comment:
Overriding the behaviour of `console.log` feels very fragile.
Instead, could we create a function called `verboseLog` which can log if verbose mode is on, and otherwise just return?

As well as being less hacky, I think it would also be less confusing to future contributors

Reply:
I wouldn't say it's fragile (as in, it wouldn't be prone to breaking), but I agree it may not be immediately obvious to contributors what's the proper way to log things.

What about creating a global logger? That may make it clearer that logging levels should be respected (while `console` usage is arguably less strict). Something like:

```
const logger = Object.create(console, {
  debug: {
    value: (message, ...args) => {
      if (userInput.verbose) {
        console.debug(message, ...args);
      }
    }
});
```

Or, to be honest, it could be as simple as the current overriding behaviour, but with the `logger` aliasing to make it clearer throughout the code:

```
const logger = console;
if (!userInput.verbose) {
  logger.debug = () => ();
}
```

Reply:
I tried implementing option 2, let me know if you want me to push it to see how it looks.

Reply:
Pushed, both the console→logger replacement and the suggestion about `--verbose` mentioned below. Waiting for feedback. 🙂

Reply:
@georgegillams,

> Overriding the behaviour of console.log feels very fragile.

Noted, but this is a small cli tool with one source code file. I think it's manageable. If it were a bigger codebase then I'd tend to agree.

Assigning `console` to `logger` just introduces even more confusion IMO. `const logger = console;` is simply assigning the _reference_ to a new variable and the act of reassigning `logger.log` is still modifying `console`.


If anything we should do this:
```
const logger = {
  debug: (...args) => {
      if (verbose) {
        console.debug(...args);
      }
  },
  ...
};
```

Reply:
I've added a commit doing something along these lines ...